### PR TITLE
Code style fixes

### PR DESCRIPTION
**Pull request by heartwilltell**

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -13,16 +13,15 @@ import (
 )
 
 const (
-	DEFAULT_INTERVAL_SECONDS = 15
+	DefaultIntervalSeconds = 15
```
Comment (heartwilltell) [inline main.go:16]: The is a convention in Go community to use camel case in constants naming.

File: main.go [hunk 2]
```diff
@@ -13,16 +13,15 @@ import (
...
 )
 
 var (
 	version string
 )
 
 func main() {
-
-	var log = log.GetLogger()
+	var logger = log.GetLogger()
```
Comment (heartwilltell) [inline main.go:24]: `log` was renamed to `logger` to prevent collision with the package name.

File: main.go [hunk 3]
```diff
@@ -77,16 +76,16 @@ func main() {
...
 	client.Init()
 
 	prometheusRegistry := prometheus.NewRegistry()
 
 	// Create each metric that we later populate
 	for _, m := range metrics.MetricsHolder {
-		prometheusRegistry.Register(m.GetPrometheusMetric())
-		client.Register(m.GetEvent(), m.GetEventHandler())
+		prometheusRegistry.Register(m.GetPrometheusMetric()) // TODO: error should be handled here
+		client.Register(m.GetEvent(), m.GetEventHandler())   // TODO: error should be handled here
```
Comment (heartwilltell) [inline main.go:88]: Errors should be handled. Always.

File: src/client/client.go [hunk 4]
```diff
@@ -34,56 +34,56 @@ func NewClient(addr string, interval int, certs TLSCerts) (*Client, error) {
...
 	if err := c.communicator.TestConnection(); err != nil {
-		log.Fatal(err)
+		logger.Fatal(err)
 		return nil, err
 	}
 	return c, nil
 }
 
-func (this *Client) Init() {
+func (c *Client) Init() {
```
Comment (heartwilltell) [inline src/client/client.go:53]: There is a convention in the Go community to name receivers with single letter. Ussually, the first letter of the type name.

File: src/client/communicator.go [hunk 5]
```diff
@@ -41,29 +40,30 @@ func (this *Communicator) TestConnection() error {
...
 	}
-	if resp.Status == "OK" {
-		return nil
-	} else {
+
+	if resp.Status != "OK" {
 		return errors.New(resp.Message)
 	}
+
+	return nil
```
Comment (heartwilltell) [inline src/client/communicator.go:48]: Simplified

File: src/client/communicator.go [hunk 6]
```diff
@@ -41,29 +40,30 @@ func (this *Communicator) TestConnection() error {
...
 
-func (this *Communicator) GetNodesData() (interface{}, error) {
+func (c *Communicator) GetNodesData() (interface{}, error) {
 	endpoint := "/api/v1/node"
 	resp := &types.NodesResponse{}
-	d, err := this.getData(endpoint, resp)
+	d, err := c.getData(endpoint, resp)
 	if err != nil {
-		return nil, fmt.Errorf("getting node data error: %s", err)
+		return nil, fmt.Errorf("getting node data error: %w", err)
```
Comment (heartwilltell) [inline src/client/communicator.go:56]: Errors should be wrapped with %w format

File: src/metrics/instance_state_per.go [hunk 7]
```diff
@@ -142,7 +142,7 @@ var ankaInstanceStatePerMetrics = []InstanceStatePerMetric{
 					}
 				}
 			}
-			checkAndHandleResetOfGuageVecMetric((len(instances) + len(instanceNodes)), "anka_instance_state_per_node_count", metric)
```
Comment (heartwilltell) [inline src/metrics/instance_state_per.go:145]: Removed unneeded parenthesis

File: src/metrics/node.go [hunk 8]
```diff
@@ -31,10 +31,10 @@ func (this NodeMetric) GetEventHandler() func(interface{}) error {
 }
 
 var ankaNodeMetrics = []NodeMetric{
-	NodeMetric{
```
Comment (heartwilltell) [inline src/metrics/node.go:34]: Removed unneeded type declaration

File: src/metrics/node_group.go [hunk 9]
```diff
@@ -11,25 +11,23 @@ type NodeGroupMetric struct {
...
 		if err != nil {
 			return err
 		}
 		// TODO: make an API call for all groups
 		var nodeGroups []types.NodeGroup
 		for _, node := range nodes { // EACH GROUP
-			for _, group := range node.Groups {
-				nodeGroups = append(nodeGroups, group)
-			}
+			nodeGroups = append(nodeGroups, node.Groups...)
```
Comment (heartwilltell) [inline src/metrics/node_group.go:27]: Simplified

File: src/metrics/node_group.go [hunk 10]
```diff
@@ -215,38 +213,38 @@ var ankaNodeGroupMetrics = []NodeGroupMetric{
 				for _, node := range nodes {
 					for _, group := range node.Groups {
 						if group.Id == focusGroup.Id {
-							count = count + node.CPUUtilization
+							count += node.CPUUtilization
```
Comment (heartwilltell) [inline src/metrics/node_group.go:216]: Simplified

**Comment by asafg6** [comment]
Hi @heartwilltell, first of all - thank you for your contribution and interest.
I'm declining this pull request since it doesn't offer any functional improvement.

**Comment by NorseGaud** [comment]
Thanks again for the PR. Unfortunately, there are far too many changes that are potentially risky for the exporter. As Asaf says, we don't accept PRs that don't add any functionality or fix a bug. I'm going to close this out for now.

**Pull request closed without merging**